src/hardware/hantek-6xxx/api.c

```diff
@@ -551,8 +551,14 @@ static uint32_t data_amount(const struct sr_dev_inst *sdi)
 		data_left = devc->samplerate * NUM_CHANNELS;
 	}
 
+#ifdef __APPLE__
+	uint32_t i = 512;
+	while (i < data_left)
+		i *= 2;
+	data_left = i;
+#else
 	data_left += MIN_PACKET_SIZE; /* Driver does not handle small buffers. */
-
+#endif
```

Comment:
Instead of this random buffer sizes we should use power of 2 ones.

http://www.cypress.com/file/138911/download
page 12/71 shows the possible endpoint buffer configurations. Currently we are using bulk on EP6 and based on the table from EP6 we can have 512 bytes quad buffer. And the firmware also sets that mode.

Reading any other buffer size looks bogus to me. And I think the reason you got errors in OSX was because the usb descriptors defined the buffer sizes and you tried to read other sizes. So set the min packet to 512 and the max packet to 1024.

I think I will try to get the firmware only running on quad 1024 buffers. That should hopefully give us 4k continuous samples at 48MHz.